apache/superset
# feat(explore): Implement viz switcher redesign

### SUMMARY
This PR implements the redesign of viz switcher. Now user can quickly change viz type to 1 of 6 featured charts without opening the viz gallery modal.
Active chart is represented by an icon and chart name. If chart's name is truncated, then tooltip appears on hover.
The rest of the featured charts are represented only by a clickable icon. Chart's name appears in a tooltip on hover. Clicking the icon switches the viz type.
User can open the viz gallery modal by clicking "View all charts". When user selects a chart that is not "featured", it's represented by a magnifying glass icon and chart's name. It's placed first on the quick viz switcher list.
When a non-featured chart is currently rendered, it's displayed at the start of the list.
Currently rendered chart has a tooltip "Currently rendered: <chart name>" 

Featured charts are:
- Echarts Line Chart
- Table
- Big Number
- Pie Chart
- Echarts Bar Chart
- Echarts Area Chart

Side note - this feature will work even better when Yongjie's [standardized form data PR](https://github.com/apache/superset/pull/20010) is merged  🙂 

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

https://user-images.githubusercontent.com/15073128/172367517-7458e8a1-3e11-404b-97a7-a541e4ad38d4.mov

### TESTING INSTRUCTIONS
1. Open explore with 1 of the featured charts
2. Verify that there are 6 icons in viz switcher, the active viz is highlighted and its name is visible
3. Click on any other icon, verify that the control panel now displays controls of currently active chart. Click Update Chart and verify that everything works as expected
4. Click "View all charts", change to a non-featured chart
5. Verify that now there are 7 items in viz switcher list and active non-featured chart is 1st on the list. Repeat step 3.

### ADDITIONAL INFORMATION



CC @kasiazjc
